proxy/src/bind.rs

```diff
@@ -32,7 +32,8 @@ use transport;
 pub struct Bind<C, B> {
     ctx: C,
     sensors: telemetry::Sensors,
-    executor: Handle,
+    // TODO: make this generic over executor?
+    executor: Sender,
```

Comment:
It's necessary to use `Sender` here rather than be generic over the executor or use `TaskExecutor::current()`. This is because the executor type is a parameter of the `tower_h2::client::Connect` struct, which means it also has to be a parameter of `transparency::client::Client`. Unfortunately, this then appears as the `Future` type of the implementation of `hyper::client::connect::Connect` for `transparency::glue::HyperConnect`, which Hyper places a `Send` bound on in the definition of `Connect`. Thus, it is necessary to use `Sender`, the only executor that is `Send`.

This is why we have to use a `ThreadPool` executor with a single thread, rather than using the `CurrentThread` executor.

Reply:
1. Within the scope of the proxy, we could maybe just assume to always use `tokio::spawn`, knowing it will use the current executor always, and we aren't a library that might need to be generic over different executors.
2. Perhaps the bounds in hyper can be loosened slightly, I'd have to look. Though if this just used `tokio::spawn`, that wouldn't break hyper's bounds, nor tie us to the threadpool.

I say this because we *may* want to use the `current_thread::Runtime` when in inject mode, and only the full thread pool in ingress mode. Not certain yet, and if we did want that, we could probably still make it work.